nupic/embodied/agents/curious_ppo_agent.py

```diff
@@ -0,0 +1,503 @@
+        Whether to apply the RewardForwardFilter and normalize rewards.
+    normadv : bool
+        Whether to normalize the advantages.
+    use_news : bool
+        Whether to take into account new episode (done=True) in the advantage
+        calculation.
+    ext_coeff : float
+        Weighting of the external rewards in the overall rewards.
+    int_coeff : float
+        Weighting of the internal rewards (disagreement) in the overall rewards.
+    nsteps_per_seg : int
+        Number of environment steps per update segment.
+    nsegs_per_env : int
+        Number of segments to collect in each environment.
+    expName : str
+        Name of the experiment (used for video logging).. currently not used
+    vLogFreq : int
+        After how many steps should a video of the training be logged.
+    dynamics_list : [Dynamics]
+        List of dynamics models to use for internal reward calculation.
+
+    Attributes
+    ----------
+    n_updates : int
+        Number of updates that were performed so far.
+    envs : [VecEnv]
+        List of vector enviornments to use for experience collection.
+
+    """
+
+    envs = None
+
+    def __init__(
+        self,
+        *,
+        scope,
+        device,
+        ob_space,
+        ac_space,
+        stochpol,
+        ent_coef,
+        gamma,
+        lam,
+        nepochs,
+        lr,
+        cliprange,
+        nminibatches,
+        normrew,
+        normadv,
+        use_news,
+        ext_coeff,
+        int_coeff,
+        nsteps_per_seg,
+        nsegs_per_env,
+        expName,
+        vLogFreq,
+        dynamics_list,
+    ):
+        self.dynamics_list = dynamics_list
+        self.n_updates = 0
+        self.scope = scope
+        self.device = device
+        self.ob_space = ob_space
+        self.ac_space = ac_space
+        self.stochpol = stochpol
+        self.nepochs = nepochs
+        self.lr = lr
+        self.cliprange = cliprange
+        self.nsteps_per_seg = nsteps_per_seg
+        self.nsegs_per_env = nsegs_per_env
+        self.nminibatches = nminibatches
+        self.gamma = gamma
+        self.lam = lam
+        self.normrew = normrew
+        self.normadv = normadv
+        self.use_news = use_news
+        self.ent_coef = ent_coef
+        self.ext_coeff = ext_coeff
+        self.int_coeff = int_coeff
+        self.vLogFreq = vLogFreq
+        # TODO: Add saving with expName & vLog Freq video saving
+
+    def start_interaction(self, env_fns, dynamics_list, nlump=1):
+        """Set up environments and initialize everything.
+
+        Parameters
+        ----------
+        env_fns : [envs]
+            List of environments, optionally with wrappers etc.
+        dynamics_list : [Dynamics]
+            List of dynamics models.
+        nlump : int
+            ..
+
+        """
+        # Specify parameters that should be optimized
+        param_list = (
+            self.stochpol.param_list + self.dynamics_list[0].auxiliary_task.param_list
```

Comment:
That is interesting. Why only the param_list of auxiliary task for the first task of the dynamic list, and not the others?